sensor/common/registry/registry_store.go

```diff
@@ -392,3 +392,221 @@ func (rs *Store) GetMatchingCentralRegistryIntegrations(imgName *storage.ImageNa
 
 	return regs
 }
+
+// UpsertSecret upserts a pull secret into the store.
+func (rs *Store) UpsertSecret(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	if !features.SensorPullSecretsByName.Enabled() {
+		rs.upsertSecretByHost(namespace, secretName, dockerConfig, serviceAcctName)
+		return
+	}
+
+	rs.upsertSecretByName(namespace, secretName, dockerConfig, serviceAcctName)
+}
+
+func (rs *Store) upsertSecretByHost(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	isGlobalPullSecret := namespace == openshiftConfigNamespace && secretName == openshiftConfigPullSecret
+	fromDefaultSA := serviceAcctName == defaultSA
+
+	for registryAddress, dce := range dockerConfig {
+		registryAddr := strings.TrimSpace(registryAddress)
+
+		if fromDefaultSA {
+			// We assume that registries found in the dockercfg secret managed by OCP for the default
+			// service account only references hostnames for the OCP internal registry.
```

Comment:
```suggestion
			// We assume registries found in the dockercfg secret managed by the OCP default
			// Service Account only references hostnames for the OCP internal registry.
```

Mostly when `account` was lowercase, I got confused thinking it was a verb and not part of the noun `service account`, so this clarifies it

Reply:
Not entirely on board with this change, will think of some better wording.  The suggestion changes the meaning and I believe it would be inaccurate. The secret is "managed by OCP", the secret is not "managed by the OCP default Service Account" (also believe lowercase is more appropriate here)

Perhaps this is better.
> Registries found in the `dockercfg` config secret associated with the `default` service account are assumed to reference the OCP internal registry.